### PR TITLE
Save all interface values if useGlobalOptions is true

### PR DESCRIPTION
**Pull request by puzzledsab**
This seemed a bit to easy so it feels like I may have missed something. I have tested it back and forth in Edge and Firefox and I can't find any problems.

Issue #1759 and #1475

### CODE REVIEW

File: sabnzbd/cfg.py [hunk 1]
```diff
@@ -85,6 +85,7 @@ def validate_server(value):
 queue_complete_pers = OptionBool("misc", "queue_complete_pers", False)
 bandwidth_perc = OptionNumber("misc", "bandwidth_perc", 100, 0, 100)
 refresh_rate = OptionNumber("misc", "refresh_rate", 0)
+interface_settings = OptionStr("misc", "interface_settings", "'{}'")
```
Comment (Safihre) [inline sabnzbd/cfg.py:88]: Is the extra `'` needed?

**Review by Safihre — CHANGES_REQUESTED** [review]
You'll need to update the tavern API tests since the output changed 🥇

File: sabnzbd/cfg.py (hunk 1 above)
Comment (puzzledsab) [reply sabnzbd/cfg.py:88]: I don't know. They're there when it saves because there are two layers of stringifying so I added them just in case.

**Comment by puzzledsab** [comment]
"'{}'" isn't '!anystr', apparently. I haven't figured out how to make this particular test run locally so I can't test it myself. Do you think !re_match "'\{.*\}'" or !re_fullmatch "'\{.*\}'" will work?

File: sabnzbd/cfg.py (hunk 1 above)
Comment (Safihre) [reply sabnzbd/cfg.py:88]: When it saves to `sabnzbd.ini`? Because the ConfigParser adds those on its own.
Does it work without the `'`?

Comment (puzzledsab) [reply sabnzbd/cfg.py:88]: The page loads just fine with no default but when new values are stored it is in the format `'{"dateFormat":"fromNow",`, so for normal usage it is there.

**Comment by jcfp** [comment]
> "'{}'" isn't '!anystr', apparently. I haven't figured out how to make this particular test run locally so I can't test it myself. Do you think `!re_match "'\{.*\}'"` or `!re_fullmatch "'\{.*\}'"` will work?

Tavern uses { and } as variable delimiters that can be escaped by doubling. Unfortunately, it keeps trying to interprete them as vars after finding a regex match, so it looks like that's a tavern bug. As a workaround, you could set strict to off/False (separate setting for json and xml, around line 5 resp. 145 in data/tavern/api_queue_format.yaml) and remove the new var from the expected output altogether for the time being.

File: interfaces/Glitter/templates/static/javascripts/glitter.main.js [hunk 2]
```diff
@@ -629,6 +629,37 @@ function ViewModel() {
         }
     })
 
+    // Save the rest in config if global-settings
+    var saveInterfaceSettings = function(newValue) {
+        if(self.useGlobalOptions()) {
+            var interfaceSettings = {
+                "dateFormat": self.dateFormat,
+                "extraQueueColumns": self.extraQueueColumns,
+                "extraHistoryColumns": self.extraHistoryColumns,
```
[the remark this answers is not in the document]
Comment (puzzledsab) [reply interfaces/Glitter/templates/static/javascripts/glitter.main.js:638]: Yes, the ko stringifier figures it all out :)

**Review by Safihre — APPROVED** [review]

**Pull request merged**